### PR TITLE
Maintenance fix

### PR DESCRIPTION
**Pull request by yerbol-akhmetov**
Good day. Here I propose changes to make the repository up-to-date with `pypsa-earth` and apply versioning. The following changes are implemented:
- [x] Fix `config.kz_default.yaml` to make it compatible with recent `pypsa-earth` release of `v0.4.0`.
- [x] Update `README.md` to instruct checking out to `v0.4.0` version of `pypsa-earth`
- [x] Tested `run_all_scenarios`, worked as intended

### CODE REVIEW

**Comment by yerbol-akhmetov** [comment]
I faced the issue Snakefile when using `run_all_scenarios` rule. Executing the scenarios one by one with `solve_everything` worked fine. The error faced when running `run_all_scenarios` is below:
image

**Comment by ekatef** [comment]
Hello @yerbol-akhmetov, great work.

Regarding the issue with `run_all_scenarios`,  can it be that merging the configs goes wrong for some reason? I'd check what is the output of the part which creates a configuration file for `run_all_scenarios` rule. That is [this part](<https://github.com/pypsa-meets-earth/pypsa-earth/blob/1bd7f84f1f9b0e081d790ffa198729076fd923b6/Snakefile#L1064-L1070>) of the code which creates `config.yaml`, when running the scenarios.

I'm thinking that it may be interesting to comment-out those two lines

```python
        os.system("snakemake -j all solve_all_networks --rerun-incomplete")
        os.system("snakemake -j1 make_statistics --force")
```

The idea is to obtain `config.yaml` and check it's structure. To find what is wrong with the structure, it may be handy to use some automated approaches, like a local run of our usual pre-commit (it also contains a linter which checks the structure of yaml files). 

What do you think?

**Comment by yerbol-akhmetov** [comment]
> Hello @yerbol-akhmetov, great work.
> 
> Regarding the issue with `run_all_scenarios`, can it be that merging the configs goes wrong for some reason? I'd check what is the output of the part which creates a configuration file for `run_all_scenarios` rule. That is [this part](https://github.com/pypsa-meets-earth/pypsa-earth/blob/1bd7f84f1f9b0e081d790ffa198729076fd923b6/Snakefile#L1064-L1070) of the code which creates `config.yaml`, when running the scenarios.
> 
> I'm thinking that it may be interesting to comment-out those two lines
> 
> ```python
>         os.system("snakemake -j all solve_all_networks --rerun-incomplete")
>         os.system("snakemake -j1 make_statistics --force")
> ```
> 
> The idea is to obtain `config.yaml` and check it's structure. To find what is wrong with the structure, it may be handy to use some automated approaches, like a local run of our usual pre-commit (it also contains a linter which checks the structure of yaml files).
> 
> What do you think?

Thank you, @ekatef. I think the issue was related to my end. I re-copied the config again, and it worked fine.

**Comment by yerbol-akhmetov** [comment]
Thank for the suggestion, @ekatef. I will add the following figure.
Kazakhstan_capacities

File: config.kz_default.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #
 # SPDX-License-Identifier: AGPL-3.0-or-later
 
-version: 1.0.0
+version: 0.3.0
```
Comment (ekatef) [inline config.kz_default.yaml]: I would probably use `version: 0.4.0` which is the config version corresponding to `0.4.0` release. Have there been any specific reasons to keep `0.3.0`?

**Review by ekatef — COMMENTED** [review]
Hello @yerbol-akhmetov, the PR looks great! 

Thanks for adding the checkout magic to use a stable version. I have only one little comment which relates to the correspondence between the config and the model versions. Could you please check it?

File: config.kz_default.yaml (hunk 1 above)
Comment (yerbol-akhmetov) [reply config.kz_default.yaml]: Thanks for noting it. I have copied the default config of `pypsa-earth` just before `v0.4.0` release. Therefore it is left as `v0.3.0`. I have modified it to match the release version.

**Review by ekatef — APPROVED** [review]
Hello @yerbol-akhmetov, looks perfect now. Well done!

**Pull request merged**